src/Network/Algorand/Node/Api.hs

```diff
@@ -181,15 +176,18 @@ data ApiAny route = ApiAny
   , _version :: route
       :- "versions"
       :> Get '[JSON] Version
-  }
-  deriving (Generic)
-
+  } deriving (Generic)
 
 -- | Algod API (v2 only).
 data ApiV2 route = ApiV2
   { _status :: route
       :- "status"
       :> Get '[JSON] NodeStatus
+  , _block :: route
+      :- "blocks"
+      :> Capture "round" Round
+      :> QueryParam "format" Text
+      :> Get '[MsgPack] BlockWrapped
```

Comment:
Actually it doesn't look nice because there is `format` parameter, but according to this line only `MsgPack` format is supported. So if someone passes `format=json`, they will always get an error here AFAIU. I think the nicest solution is to have `'[MsgPack, JSON]` here and define JSON instances for `BlockWrapped`. Otherwise, we should at least add a comment saying "do not try passing format other than `msgpack` here".

Reply:
I've used `$(deriveJSON defaultOptions 'Block)`
not sure that `defaultOptions` is good.. what do you think?

Reply:
I don't think it will work with `defaultOptions`. You can test it by passing `json` as `format` parameter. I suspect it won't be parsed successfully. I see there is `Data.Algorand.MessagePack.Json` and it's used in some places to define `ToJSON` and `FromJSON` instances. I would try using functions from that module.

Reply:
Oh! there are no options which will work

cause it returns 
```
{
  "block": {
    "fees": "A7NMWS3NT3IUDMLVO26ULGXGIIOUQ3ND2TXSER6EBGRZNOBOUIQXHIBGDE",
    "frac": 250000000,
    "gen": "testnet-v1.0",
    "gh": "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=",
    "nextbefore": 10001,
    "nextproto": "https://github.com/algorand/spec/tree/22726c9dcd12d9cddce4a8bd7e8ccaa707f74101",
    "nextswitch": 20001,
    "nextyes": 1,
    "prev": "blk-DPCGNQFB3PZCIGUG6MBL6ASVM3RD33UXJX65XURCJL5WIIRUAJJQ",
    "proto": "https://github.com/algorand/spec/tree/a26ed78ed8f834e2b9ccb6eb7d3ee9f629a6e622",
    "rate": 250000000,
    "rnd": 1,
    "rwcalr": 500000,
    "rwd": "7777777777777777777777777777777777777777777777777774MSJUVU",
    "seed": "sBoIA83cUpa8jNOdqmGClw85y0bYcFhfmKzrnxmhgdc=",
    "ts": 1560210480,
    "txn": "LN9CHN+zm1OHdoF2SPpftJW9YI2J/F0Zv/FJsNdAheA=",
    "txns": [
      {
        "hgh": true,
        "hgi": true,
        "sig": "tlAccbsxpTazfxt3Yu69Li98QvH6nDAPNHdU8LUkpBLxu1umNePq0NwfHbrtv3hW4dxKQw32SbszeMrv5X64Cg==",
        "txn": {
          "amt": 100000000,
          "fee": 10000,
          "lv": 1000,
          "rcv": "3NVE2MK2QYZQFOZ5XIRQTM7JRHNPUBV7QKLYLT7OO6QXFHXMRIAUXXNCBM",
          "snd": "GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A",
          "type": "pay"
        }
      }
    ],
    "upgradeprop": "https://github.com/algorand/spec/tree/22726c9dcd12d9cddce4a8bd7e8ccaa707f74101",
    "upgradeyes": true
  }
}
```

and we have completely different field names in our records...

Reply:
We already use these names for messagepack encoding, we just need to also use them in JSON. AFAIU `Data.Algorand.MessagePack.Json` is supposed to handle that.

Reply:
We already use these names for messagepack encoding, we just need to also use them in JSON. AFAIU `Data.Algorand.MessagePack.Json` is supposed to handle that.

Reply:
It's not trivial and requires to define a lot of instances.. I have stashed changes and we can continue later, but first - let's finish with #1